js/app.js

```diff
@@ -1,122 +1,132 @@
-          Project.loadAll(JSON.stringify(data));
-          Project.appendAll(projects);
-          //Set Local Storage
-          localStorage.setItem('data', JSON.stringify(projects));
-          console.log('Setting Local Data...');
+  Project.fetchAll = function(){
+    if (localStorage.data) {
+      console.log('Local Storage DOES Exist');
+
+      $.ajax({
+        type: 'head',
+        url: 'data/projects-data.json',
+        success: function (data, message, xhr) {
+          var eTag = xhr.getResponseHeader('eTag');
+          console.log('Current eTag = ' + eTag);
+          var compareETag = JSON.parse(localStorage.getItem('eTag1'));
+          console.log('Compare eTag = ' + compareETag);
+
+          if (eTag !== compareETag) {
+            console.log('Changes to JSON file detected');
+            console.log('Retrieved data from JSON file...');
+            Project.loadAll(JSON.stringify(data));
+            Project.appendAll(projects);
+            //Set Local Storage
+            localStorage.setItem('data', JSON.stringify(projects));
+            console.log('Setting Local Data...');
+            Project.appendAll(projects);
+          }
+          else {
+            console.log('NO change to JSON Data');
+            var localData = localStorage.getItem('data');
+            console.log('Retriving Local Data');
+            Project.loadAll(localData);
+          }
           Project.appendAll(projects);
         }
-        else {
-          console.log('NO change to JSON Data');
-          var localData = localStorage.getItem('data');
-          console.log('Retriving Local Data');
-          Project.loadAll(localData);
-          //Append to HTML
-        }
+      });
+    }
+    else {
+      $.getJSON('data/projects-data.json', function(data){
+        console.log('Retrieved data from JSON file...');
+        Project.loadAll(JSON.stringify(data));
         Project.appendAll(projects);
-      }
-    });
-  }
-  else {
-    $.getJSON('data/projects-data.json', function(data){
-      console.log('Retrieved data from JSON file...');
-      Project.loadAll(JSON.stringify(data));
-      Project.appendAll(projects);
-      //Set Local Storage
-      localStorage.setItem('data', JSON.stringify(projects));
-      console.log('Locale Storage DOES NOT exist.');
-      console.log('Setting Local Data...');
-    }).success(function(data, message, xhr) {
-      var eTag1 = xhr.getResponseHeader('eTag');
-      localStorage.setItem('eTag1', JSON.stringify(eTag1));
-    });
-  }
-};
+        localStorage.setItem('data', JSON.stringify(projects));
+        console.log('Locale Storage DOES NOT exist.');
+        console.log('Setting Local Data...');
+      })
+      .success(function(data, message, xhr) {
+        var eTag1 = xhr.getResponseHeader('eTag');
+        localStorage.setItem('eTag1', JSON.stringify(eTag1));
+      });
+    }
+  };
 
+  // Call Functions
+  $(document).ready(function(){
```

Comment:
Now we're doing functional programming, it would be good if you could abstract these out into a function you could call...some sort of projectView.init?